xray/cli.go

```diff
@@ -114,3 +145,81 @@ func newXrCurlCommand(c *cli.Context) (*curl.XrCurlCommand, error) {
 	xrCurlCommand.SetUrl(xrDetails.XrayUrl)
 	return xrCurlCommand, err
 }
+
+func auditNpmCmd(c *cli.Context) error {
+	serverDetailes, err := CreateServerDetailsWithConfigOffer(c)
+	if err != nil {
+		return err
+	}
+	var typeRestriction = npmutils.All
+	switch c.String("type-restriction") {
+	case "devOnly":
+		typeRestriction = npmutils.DevOnly
+	case "prodOnly":
+		typeRestriction = npmutils.ProdOnly
+	}
+	auditNpmCmd := audit.NewAuditNpmCommand().SetServerDetails(serverDetailes).SetNpmTypeRestriction(typeRestriction).SetTargetRepoPath(c.String("repo-path"))
+	// Handle Xray's contex flags
+	if c.String("watches") != "" {
+		auditNpmCmd.SetWatches(strings.Split(c.String("watches"), ","))
+	} else if c.String("project") != "" {
+		auditNpmCmd.SetProject(c.String("project"))
+	}
+	auditNpmCmd.SetIncludeVulnerabilities(shouldIncludeVulnerabilities(c))
+	return commands.Exec(auditNpmCmd)
+}
+
+func scanCmd(c *cli.Context) error {
+	serverDetailes, err := CreateServerDetailsWithConfigOffer(c)
+	if err != nil {
+		return err
+	}
+	var specFile *spec.SpecFiles
+	if c.IsSet("spec") {
+		specFile, err = cliutils.GetFileSystemSpec(c)
+	} else {
+		specFile, err = createDefaultScanSpec(c, c.String("repo-path"))
```

Comment:
What si the reason we're setting the "repo-path" as the file spec target? This file spec doesn't upload fikes.
Also, shouldn't we send this "repo-path" value to Xray, similarly to "watches" or "project"?